src/backend/distributed/operations/repair_shards.c

```diff
@@ -529,7 +529,7 @@ ErrorIfMoveUnsupportedTableType(Oid relationId)
  * list. The function assumes that all the shards in the list are colocated.
  */
 void
-BlockWritesToShardList(List *shardList)
+BlockWritesToColocatedShardList(List *shardList)
```

Comment:
Hmm, I think this change is wrong? It should be `BlockWritesToShardList` because the input shard list already contains the colocated shards, this simply iterates over the list

Reply:
I discussed this with @marcocitus and he suggested this name.
I didn't understand why you thought the `BlockWritesToShardList` is better if the list is already colocated.

Reply:
Do you mean it sounds like it is going to block the writes to the shards that are colocated to the ones at the list?

Reply:
> Do you mean it sounds like it is going to block the writes to the shards that are colocated to the ones at the list?

Yes, it sounds like that to me. Plus, it seems nicer to make that in a separate PR (or at least in a separate commit).

And, it makes hard to merge enterprise